src/server/json_family.cc

```diff
@@ -634,8 +643,112 @@ OpResult<vector<OptString>> OpArrPop(const OpArgs& op_args, string_view key, str
   return vec;
 }
 
+// Returns string vector that represents the new length of the array at each path.
+OpResult<vector<OptSizeT>> OpArrInsert(const OpArgs& op_args, string_view key, string_view path,
+                                       int index, vector<json> new_values) {
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return result.status();
+  }
+
+  bool out_of_boundaries_encountered = false;
+  vector<OptSizeT> vec;
+  auto cb = [&](const string& path, json& val) {
```

Comment:
could you add comments explaining what your function does at high level?

Reply:
Sure.